src/messages.lisp

```diff
@@ -276,6 +276,30 @@
       :required nil))
   :documentation "Native Quil and associated metadata returned from quilc.")
 
+(defmessage |RewriteArithmeticRequest|
+    ((|quil|
+      :documentation "Native Quil for which to rewrite arithmetic parameters."
+      :type :string
+      :required t))
+  :documentation "A request type to handle compiling arithmetic out of gate parameters.")
+
+(defmessage |RewriteArithmeticReponse|
+    ((|quil|
+      :documentation "Native Quil rewritten with no arithmetic in gate parameters."
+      :type :string
+      :required t)
+
+     (|original_memory_descriptors|
+      :documentation "The declared memory descriptors in the Quil of the related request."
+      :type (:list :any)
+      :required nil)
+
+     (|recalculation_table|
+      :documentation "A mapping from memory references to the original gate arithmetic."
+      :type (:map :any)
```

Comment:
I'd prefer that this have a more descriptive type. (In fact, I'm surprised that `(:map :any)` is a legal type.) From conversation it sounds like it should be `(:map |ParameterAref| -> :string)`, but I'm not confident in this.

Reply:
I think that is what it will be when it's serialized properly, as you mentioned